modules/e2ee/Worker.js

```diff
@@ -52,15 +63,22 @@ onmessage = async event => {
         const context = getParticipantContext(participantId);
 
         handleTransform(context, operation, readableStream, writableStream);
-    } else if (operation === 'setKey') {
+    } else if (operation === 'setKeyBytes') {
```

Comment:
This looks almost a carbon copy of setKey. How about this: a subclass of Context, called SharedContext, which overrides setKey, does the needful, and calls _setKey.